msal/confidential_client_application.go

```diff
@@ -4,10 +4,31 @@
 package msal
 
 import (
+	"context"
+	"net/http"
+
 	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/msalbase"
 	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/requests"
 )
 
+// ConfidentialClientApplicationOptions configures the PublicClientApplication's behavior.
+type ConfidentialClientApplicationOptions struct {
```

Comment:
A few things I'd note here:
- We really need to do something about these names (though probably not in this PR)
- We should not use the options struct. This is non-go standard.  I know the official SDK track 2 guidelines say to do this, but since we aren't in the SDK, we should take advantage of variardic options.  This struct would become private and modified by those args.  It handles better defaults.
- HTTPClient, which we are renaming somewhere above, should just be eliminated in the end.  If we really have a use case, we can add it back later (but I doubt we will).

Reply:
The Go blog details both variadic config and optional args struct as cromulent ways for handling configuration.  There's a nice write-up as part of the module compatibility blog post.

https://blog.golang.org/module-compatibility

We (the Go SDK team) went round and round on this, and in the end we felt that the self-documenting properties of the optional params struct is the better customer experience.  Note that we haven't had any usability studies as of yet, so our position may change.

Having a configurable HTTP client allows the use of something like our pipeline middleware for controlling request/response lifetime.  Most common case I the ability to specify a retry policy for handling transient failures.  At any rate we can defer it for now.

Reply:
I'm not saying it has to be optional args struct, just providing my perspective (whileI do think there's benefit of having consistency across SDKs, this isn't the hill I'm gonna die on).

Reply:
We have bring your own HttpClient in all our libraries. Has been a wish so many times that we are providing it out of the box with new libraries we write.